### DISCUSSION TITLE
Can you tell when share() completes/unsubscribes?

### DISCUSSION THREAD

**Discussion by raymonddavis** [ReactiveX/rxjs#6253] [Q&A] [answered]
The share() pipe completes when the last subscriber unsubscribes. Is there a way to be notified or run a function when the share unsubscribes?

**Comment by cartant** [comment, marked as answer]
You can use the `finalize` operator to run a function when the source completes, errors or is explicitly unsubscribed:

```ts
const shared = source.pipe(
   finalize(() => { /* whatever */ }),
   share()
);
```
